snap/implicit.go

```diff
@@ -139,6 +139,11 @@ func addImplicitHooksFromContainer(snapInfo *Info, snapf Container) error {
 }
 
 func addHookName(snapInfo *Info, hookName string) {
+	// Verify that the hook name is actually supported. If not, ignore it.
+	if !IsHookSupported(hookName) {
```

Comment:
It'd be slightly nicer to have this check done at the call site on line 135 instead. It's surprising an error prone to say doSomething and then just ignore it altogether. I'd expect an error, or a panic, but not complete silence and refusal to do what it's supposed to.